File: frontend/src/app/core/services/paiement.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import {
  Paiement,
  PaiementRequest,
  PaiementResponse
} from '../models/paiement.model';

@Injectable({
  providedIn: 'root'
})
export class PaiementService {
  private apiUrl = `${environment.apiUrl}/api/paiements`;

  constructor(private http: HttpClient) {}

  getPaiements(): Observable<PaiementResponse[]> {
    return this.http.get<PaiementResponse[]>(this.apiUrl);
  }

  getPaiement(id: number): Observable<PaiementResponse> {
    return this.http.get<PaiementResponse>(`${this.apiUrl}/${id}`);
  }

  getPaiementsByFacture(factureId: number): Observable<PaiementResponse[]> {
    return this.http.get<PaiementResponse[]>(`${this.apiUrl}/facture/${factureId}`);
  }

  processPaiement(request: PaiementRequest): Observable<PaiementResponse> {
    return this.http.post<PaiementResponse>(this.apiUrl, request);
  }
}

File: backend/paiementservice/src/main/java/com/example/paiementservice/service/PaiementService.java
package com.example.paiementservice.service;

import com.example.paiementservice.dto.PaiementRequest;
import com.example.paiementservice.dto.PaiementResponse;

import java.util.List;
import java.util.Optional;

public interface PaiementService {
    List<PaiementResponse> getAllPaiements();
    Optional<PaiementResponse> getPaiementById(Long id);
    List<PaiementResponse> getPaiementsByFactureId(Long factureId);
    PaiementResponse processPaiement(PaiementRequest request);
}

